src/main/java/frc/robot/subsystems/Vision/VisionHelper.java

```diff
@@ -0,0 +1,423 @@
+import edu.wpi.first.math.Nat;
+import edu.wpi.first.math.geometry.Pose3d;
+import edu.wpi.first.math.geometry.Quaternion;
+import edu.wpi.first.math.geometry.Rotation3d;
+import edu.wpi.first.math.geometry.Transform3d;
+import edu.wpi.first.math.geometry.Translation3d;
+import edu.wpi.first.math.numbers.N1;
+import edu.wpi.first.math.numbers.N3;
+import edu.wpi.first.math.numbers.N5;
+import edu.wpi.first.math.util.Units;
+import edu.wpi.first.wpilibj.DriverStation;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import org.littletonrobotics.junction.LogTable;
+import org.photonvision.EstimatedRobotPose;
+import org.photonvision.PhotonPoseEstimator.PoseStrategy;
+import org.photonvision.estimation.TargetModel;
+import org.photonvision.estimation.VisionEstimation;
+import org.photonvision.targeting.PhotonPipelineResult;
+import org.photonvision.targeting.PhotonTrackedTarget;
+import org.photonvision.targeting.TargetCorner;
+
+public class VisionHelper {
+  // constants
+  public static final Matrix<N3, N3> CAMERA_MATRIX_OPT =
+      MatBuilder.fill(
+          Nat.N3(),
+          Nat.N3(),
+          923.5403619629557,
+          0.0,
+          644.4965658066068,
+          0.0,
+          925.8136962361125,
+          402.6412935350414,
+          0.0,
+          0.0,
+          1.0); // TODO tune maybe
+  public static final Matrix<N5, N1> DIST_COEFFS_OPT =
+      MatBuilder.fill(
+          Nat.N5(),
+          Nat.N1(),
+          0.05452153950284706,
+          -0.04331612051891956,
+          0.00176988756858703,
+          -0.004530368741385627,
+          -0.040501622476628085); // TODO tune maybe
+  public static final Transform3d CAMERA_TO_ROBOT = new Transform3d(); // TODO find
+  public static final List<TagCountDeviation> TAG_COUNT_DEVIATION_PARAMS =
+      List.of(
+          // 1 tag
+          new TagCountDeviation(
+              new UnitDeviationParams(.25, .4, .9),
+              new UnitDeviationParams(.35, .5, 1.2),
+              new UnitDeviationParams(.5, .7, 1.5)),
+
+          // 2 tags
+          new TagCountDeviation(
+              new UnitDeviationParams(.35, .1, .4), new UnitDeviationParams(.5, .7, 1.5)),
+
+          // 3+ tags
+          new TagCountDeviation(
+              new UnitDeviationParams(.25, .07, .25), new UnitDeviationParams(.15, 1, 1.5)));
+
+  public static void logPhotonTrackedTarget(
+      PhotonTrackedTarget target, LogTable table, String name) {
+
+    logTransform3d(target.getBestCameraToTarget(), table, name);
+    logTransform3d(target.getAlternateCameraToTarget(), table, "alt " + name);
+    logCorners(target, table, name);
+
+    table.put("yaw " + name, target.getYaw());
+    table.put("pitch " + name, target.getPitch());
+    table.put("area " + name, target.getArea());
+    table.put("skew " + name, target.getSkew());
+    table.put("fiducial id " + name, target.getFiducialId());
+    table.put("pose ambiguity " + name, target.getPoseAmbiguity());
```

Comment:
probably should have consistent capitalization schemes with the other logged fields but tbh doesnt matter